Sources/Fuzzilli/FuzzIL/ProgramTypes.swift

```diff
@@ -73,7 +69,63 @@ public struct ProgramTypes: Equatable, Sequence {
         return types[variable]?.last(where: { $0.index <= instrIndex })?.type ?? .unknown
     }
 
-    public func makeIterator() -> VariableMap<[TypeIndexPair]>.Iterator {
+    // Filter out only runtime types
+    public func getRuntimeTypes() -> ProgramTypes {
+        var runtimeTypes = ProgramTypes()
+        for (variable, instrTypes) in types {
+            for typeInfo in instrTypes {
+                guard typeInfo.quality == .runtime else { continue }
+
+                runtimeTypes.setType(
+                    of: variable, to: typeInfo.type, at: typeInfo.index, quality: .runtime
+                )
+            }
+        }
+        return runtimeTypes
+    }
+
+    // Format ProgramTypes struct so searching for type changes at instruction is easier
+    public func getInvertedStructure(for program: Program) -> [[(Variable, Type)]] {
```

Comment:
Maybe name it `indexedByInstruction` or so, since then it's a bit clearer from the name what this actually does?